src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,160 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> lastNode;
+    private Node<T> firstNode;
+
     @Override
     public void add(T value) {
+        addLastNode(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            checkIndex(index);
+            addNodeInside(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T node : list) {
+            add(node);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T oldValue = get(index);
+        getNode(index).item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return unlink(getNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> temporaryNode = firstNode; temporaryNode != null;
+                    temporaryNode = temporaryNode.next) {
+                if (temporaryNode.item == null) {
+                    unlink(temporaryNode);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> temporaryNode = firstNode; temporaryNode != null;
+                    temporaryNode = temporaryNode.next) {
+                if (object.equals(temporaryNode.item)) {
+                    unlink(temporaryNode);
+                    return true;
+                }
+            }
+        }
```

Comment:
i think we can simplify. public boolean remove(T object) {
for (Node<T> temporaryNode = firstNode; temporaryNode != null;
                    temporaryNode = temporaryNode.next) {
                     if (temporaryNode.item == object || (temporaryNode.item != null && object.equals(temporaryNode.item))) {....